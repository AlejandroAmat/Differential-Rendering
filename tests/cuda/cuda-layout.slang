//TEST(compute):COMPARE_COMPUTE_EX:-cuda -compute -output-using-type -shaderobj

//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer : register(u0);

#define WRITE_TYPE_ALIGN(base, type) \
    outputBuffer[base * 4 + 0] = __alignOf<type>(); \
    outputBuffer[base * 4 + 1] = __alignOf<vector<type, 2> >(); \
    outputBuffer[base * 4 + 2] = __alignOf<vector<type, 3> >(); \
    outputBuffer[base * 4 + 3] = __alignOf<vector<type, 4> >();
    

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    WRITE_TYPE_ALIGN(0, uint8_t)
    WRITE_TYPE_ALIGN(1, uint16_t)
    WRITE_TYPE_ALIGN(2, int)
    WRITE_TYPE_ALIGN(3, int64_t)
    
    WRITE_TYPE_ALIGN(4, half)
    WRITE_TYPE_ALIGN(5, float)
    WRITE_TYPE_ALIGN(6, double)   
}
