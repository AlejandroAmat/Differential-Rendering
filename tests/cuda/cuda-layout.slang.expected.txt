type: int32_t
1
2
1
4
2
4
2
8
4
8
4
16
8
16
8
16
2
4
4
4
4
8
4
16
8
16
8
16
0
0
0
0
